Sprint 14

DmitryBeton opened the pull request.

DmitryBeton committed:
feat: добавлена базовая структура моделей трекеров

- Добавлена структура Tracker с поддержкой регулярных и нерегулярных трекеров
- Добавлена структура TrackerCategory для группировки трекеров
- Добавлена структура TrackerRecord для отслеживания выполненных трекеров
- Добавлена структура Schedule для управления расписанием

DmitryBeton committed:
refactor: перевести проект на архитектуру MVP

- Разделить ответственность между View, Presenter
- Добавить протоколы для всех компонентов
- Реорганизовать структуру папок по модулям
- Добавить Mock сервисы для тестирования

DmitryBeton committed:
feat(trackers): добавлены заголовки секций и проверка отметки трекеров на будущие даты

- Добавлен TrackerHeaderView для отображения названий категорий в collection view
- Настроен layout и отступы коллекции
- Добавлена проверка даты в презентере — нельзя отмечать трекеры на будущие даты
- Добавлено уведомление (alert) при попытке отметить трекер в будущем

DmitryBeton committed:
fix(trackers): устранить наложение ячеек при отображении пустого состояния

- Очистить коллекцию перед показом пустого состояния
- Обеспечить корректное скрытие элементов коллекции при отсутствии трекеров

DmitryBeton committed:
feat(trackers): обновил UI TrackersViewController и добавил заготовку для CreateTracker

DmitryBeton committed:
fix(trackers): устранил проблему UI в TrackerCollectionViewCell — скорректировано склонение слова "день"

DmitryBeton committed:
feat(create-tracker): добавлен новый экран создания трекера через present

- Создан CreateTrackerViewController
- Добавлено текстовое поле для ввода названия трекера
- Добавлена таблица с пунктами «Категории» и «Расписание»
- Настроены закругленные ячейки и отключение скролла таблицы

DmitryBeton committed:
feat(schedule): добавить экран выбора дней недели для трекера

- Создан новый экран ScheduleViewController для настройки расписания трекера
- Реализован интерфейс выбора дней недели с использованием UITableView
- Добавлены переключатели (UISwitch) для каждого дня недели

DmitryBeton committed:
feat: реализация создания трекера  - По рекомендации наставника NaR0RoW(GitHub), добавлен SwiftLog
- CreateTrackerViewController: валидация текстового поля, отображение расписания, управление состоянием кнопки
- ScheduleViewController: выбор дней с переключателями, callback через делегат
- TrackersPresenter: добавлен метод createNewTracker для обработки новых трекеров
- MockTrackersRepository: реализован метод addTracker
- Расширение TrackerSchedule свойством displayText для UI
- Добавлен UITapGestureRecognizer для скрытия клавиатуры

DmitryBeton committed:
fix: добавлено скрытие клавиатуры по кнопке Done

- Добавлен UITextFieldDelegate в CreateTrackerViewController
- Реализован метод textFieldShouldReturn для скрытия клавиатуры

DmitryBeton committed:
fix(schedule): Задана верная высота ячеек на экране ScheduleViewController

DmitryBeton committed:
refactor: прожата комбинация ctrl + i во всех файлах

d00m1r commented on Tracker/App/Resources/Assets.xcassets/Images/AddTracker.imageset/Add tracker.png:
Картинки 1x лучше не добавлять. Они нужны были для iPhone без retina экранов, а последний такой - iPhone 4. Он не поддерживает минимальную версию приложения, поэтому можно смело удалять, чтобы не занимать лишнюю память

d00m1r commented on Tracker/Modules/Trackers/Data/TrackerRepositoryMock.swift:
```diff
@@ -0,0 +1,59 @@
+        return categories.compactMap { category in
+            let filtered = category.trackers.filter { tracker in
+                guard let schedule = tracker.schedule else { return true }
+                switch weekday {
```
Можно указать только 1 `return` перед `switch`
```swift
return switch weekday {
case 1: schedule.sunday
...
}
```

DmitryBeton committed:
fix: ⚠️ Критические замечания исправлены:

– Иконка добавления соответствует дизайн-макету (пункт 11)
– Шрифт шапки и ячеек на экране создания трекера соответствуют дизайн-макету (пункт 11)
– Цвет тогглов и шрифт шапки на экране выбора расписаний соответствуют дизайн-макету (пункт 11)
– Изображение заглушки центрировано (пункт 11)

DmitryBeton committed:
fix: задан правильный горизонтальный отступ у header

DmitryBeton committed:
fix(trackers): Кнопка добавление трекера корректно изменяет цвет в зависимости от темы приложения

DmitryBeton committed:
fix(create-tracker): Добавлено ограничение 38 символов для UITextField
